src/client-agent/start_agent.c

```diff
@@ -233,8 +233,9 @@ static void w_agentd_keys_init (void) {
             merror_exit(AG_NOKEYS_EXIT);
         }
     }
-
-    OS_StartCounter(&keys);
+    else {
+        OS_StartCounter(&keys);
```

Comment:
Maybe, we could a comment here explaining where is `OS_StartCounter()` being called if  `keys.keysize == 0`

Reply:
I think to add this comment here too. But, in my opinion, it's not recommended to link what happens in other parts of the code as they can change and is difficult to sustain.
I preferred to keep this simple as:
With keys: Initialize.
Without keys: Start a more complex process that covers a key request and other initializers.
Otherwise, we are not explaining that with keys we will re-read the complete keys in example.

But, if you consider it is difficult to understand the reason for the else statement, we can add this comment. Sure.

Reply:
I added a comment as suggested.

Reply:
Thank you!